src/common/ServerAPI.jsx

```diff
@@ -30,13 +32,53 @@ async function loginAccount(email, password) {
   const masterKey = await deriveMasterKey(email, password);
   const passwordHash = await deriveMasterPasswordHash(password, masterKey);
 
-  return axios.post(
-    LOGIN_ACCOUNT_URL,
-    {
-      email: email,
-      password: passwordHash,
-    },
-  );
+  try {
+      const response = await axios.post(
+          LOGIN_ACCOUNT_URL,
+          {
+              email: email,
+              password: passwordHash,
+          },
+      );
+
+      const encryptedSymmetricKey = response.data.encrypted_symmetric_key;
+      const symmetricKey = await decryptSymmetricKey(encryptedSymmetricKey, masterKey);
+
+      return Promise.resolve(symmetricKey);
```

Comment:
Looking closer at these functions, I don't need the try..catch blocks at all, because the functions are themselves called inside more meaningful try..catch blocks (in `LoginForm.jsx` and `DashboardPage.jsx`)

Reply:
I see what you mean. The error doesn't need to be handled here and should be handled higher up the call stack where this function is being called. Also the error gets propagated up the stack anyways with Promise.reject so it's unnecessary.